azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -435,31 +399,148 @@ def test_send_method_response_calls_pipeline(self, client, pipeline):
+    def test_patch_twin_reported_properties_calls_pipeline(self, client, pipeline):
+        reported_properties = {"key": "value"}
+        client.patch_twin_reported_properties(reported_properties)
+        assert pipeline.patch_twin_reported_properties.call_count == 1
+        assert pipeline.patch_twin_reported_properties.call_args[1]["patch"] is reported_properties
 
-        response = MethodResponse(request_id="1", status=200, payload={"key": "value"})
-        client.send_method_response(response)
+    @pytest.mark.it(
+        "Waits for the completion of the 'send_method_response' pipeline operation before returning"
+    )
+    def test_waits_for_pipeline_op_completion(self, mocker, client_manual_cb, pipeline_manual_cb):
+        self.add_event_completion_checks(
+            mocker=mocker, pipeline_function=pipeline_manual_cb.patch_twin_reported_properties
+        )
+        reported_properties = {"key": "value"}
+        client_manual_cb.patch_twin_reported_properties(reported_properties)
+
+
+class SharedClientReceiveTwinDesiredPropertiesPatchTests(object):
+    @pytest.fixture
+    def patch(self):
+        return {"properties": {"desired": {"foo": 1}}}
+
+    @pytest.mark.it(
+        "Implicitly enables Twin desired properties patch feature if not already enabled"
+    )
+    def test_enables_twin_patches_only_if_not_already_enabled(self, mocker, client, pipeline):
+        mocker.patch.object(
+            SyncClientInbox, "get"
+        )  # patch this so receive_twin_desired_properties_patch won't block
+
+        # Verify twin patches enabled if not enabled
+        pipeline.feature_enabled.__getitem__.return_value = (
+            False
+        )  # twin patches will appear disabled
+        client.receive_twin_desired_properties_patch()
+        assert pipeline.enable_feature.call_count == 1
+        assert pipeline.enable_feature.call_args[0][0] == constant.TWIN_PATCHES
+
+        pipeline.enable_feature.reset_mock()
+
+        # Verify twin patches not enabled if already enabled
+        pipeline.feature_enabled.__getitem__.return_value = True  # C2D will appear enabled
+        client.receive_twin_desired_properties_patch()
+        assert pipeline.enable_feature.call_count == 0
+
+    @pytest.mark.it("Returns a message from the twin patch inbox, if available")
+    def test_returns_message_from_twin_patch_inbox(self, mocker, client, patch):
+        inbox_mock = mocker.MagicMock(autospec=SyncClientInbox)
+        inbox_mock.get.return_value = patch
+        manager_get_inbox_mock = mocker.patch.object(
+            client._inbox_manager, "get_twin_inbox", return_value=inbox_mock
+        )
+
+        received_patch = client.receive_twin_desired_properties_patch()
+        assert manager_get_inbox_mock.call_count == 1
+        assert inbox_mock.get.call_count == 1
+        assert received_patch is patch
+
+    @pytest.mark.it("Can be called in various modes")
```

Comment:
>Can be called in various modes [](start = 21, length = 30)

This description is not very clear at first glance. Once you see parameters it is more clearer

Reply:
Given that it will always render with parameters I think that's acceptable - too specific and it won't render well with the parameters.